### PR TITLE
Sinon.JS Added missing methods

### PR DESCRIPTION
**Pull request by johan-gorter**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://sinonjs.org/releases/v2.1.0/fake-xhr-and-server/
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

File: types/sinon/index.d.ts [hunk 1]
```diff
@@ -79,6 +79,7 @@ declare namespace Sinon {
         alwaysReturned(obj: any): boolean;
         invokeCallback(...args: any[]): void;
         getCall(n: number): SinonSpyCall;
+        getCalls(): SinonSpyCall[];
```
Comment (rationull) [inline types/sinon/index.d.ts]: I don't see any documentation on sinonjs.org for a getCalls() on the spy API, only getCall(). Can you provide a reference for this?

I do see it's included in the Sinon source code, but I don't think that undocumented or private APIs are supposed to be included in typings. So this change might require clarification from the Sinon developers and a doc change before being accepted.

Anyone more deeply involved with DefinitelyTyped please correct me on this if I'm wrong about what should be accepted!

**Comment by rationull** [comment]
I had one request for feedback on a changed file, and also I think it would be good to include test coverage for the new additions but that's minor and probably not required as these are simple changes.

I'm not sure why the CI build failed. The failure is in sinon-test which this PR did not change. Pre-existing problem related to changes to linting requirements?

**Comment by andy-ms** [comment]
Test fixed by #15957, please rebase.

File: types/sinon/index.d.ts (hunk 1 above)
Comment (andy-ms) [inline types/sinon/index.d.ts]: In a well-documented package you shouldn't include something that's not documented and may change.

**Comment by johan-gorter** [comment]
I wasn't aware that getCalls() is undocumented. We rely heavily on it in our codebase with code like `mouseEnterListener = addEventListener.getCalls().find(stub => stub.args[0] === 'mouseenter')`

I opened a pull request with Sinon.JS to get this documented: https://github.com/sinonjs/sinon-docs/pull/41

**Comment by rationull** [comment]
To be fair it does seem most likely by far that this is just a doc oversight as it wouldn't make sense for getCalls to be a private API function.

An addition to the associated test file would at least cause CI to show a problem if getCalls were not present.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**